torch/testing/_internal/common_utils.py

```diff
@@ -2153,6 +2196,21 @@ def __init__(self, method_name='runTest'):
             if self._ignore_not_implemented_error:
                 self.wrap_with_policy(method_name, lambda: skip_exception_type(NotImplementedError))
 
+            if PRINT_REPRO_ON_FAILURE:
+                env_var_prefix = get_test_feature_env_var_prefix()
+                try:
+                    test_filename = os.path.relpath(inspect.getfile(type(self)), start=os.curdir)
+                    repro_str = f"""
+To execute this test, run the following from the base repo dir:
+    {env_var_prefix} python {test_filename} -k {method_name}
+                    """
+                    self.wrap_with_policy(
+                        method_name,
+                        lambda repro_str=repro_str: print_repro_on_failure(repro_str=repro_str))
+                except Exception:
+                    # Don't fail entirely if we can't get the test filename
+                    pass
```

Comment:
worth logging the exception here, `log.exception("could not print repro string")`